test/testmachinery/system/shoot_cp_migration/migrate_test.go

```diff
@@ -181,12 +181,20 @@ func initGuestBookTest(ctx context.Context, t *ShootMigrationTest) (*application
 		ShootClient:       t.ShootClient,
 		SeedClient:        t.SourceSeedClient,
 	}
-	if !t.Shoot.Spec.Addons.NginxIngress.Enabled {
+	if t.Shoot.Spec.Addons == nil || t.Shoot.Spec.Addons.NginxIngress == nil || !t.Shoot.Spec.Addons.NginxIngress.Enabled {
 		if err := t.GardenerFramework.UpdateShoot(ctx, &t.Shoot, func(shoot *gardencorev1beta1.Shoot) error {
 			if err := t.GardenerFramework.GetShoot(ctx, shoot); err != nil {
 				return err
 			}
 
+			if shoot.Spec.Addons == nil {
+				shoot.Spec.Addons = &gardencorev1beta1.Addons{}
+			}
+
+			if shoot.Spec.Addons.NginxIngress == nil {
+				shoot.Spec.Addons.NginxIngress = &gardencorev1beta1.NginxIngress{}
+			}
```

Comment:
@plkokanov do you recall why we need this and why https://github.com/gardener/gardener/blob/1f8712f7cd50b7cc023ecea12c0db75c5b9f4ec1/test/framework/applications/guestbooktest.go#L132-L134 is not enough?

If you run the guestbook test against a Shoot with the addon disabled, the test will fail with meaningful error message. Can't we do something similar here?

Reply:
Well, I think it could have had failed with the mentioned error. 
In my scenario I was running the test locally with environment basically build with the command `make kind-up; make gardener-up; make kind2-up; make gardenlet-kind2-up`. The shoot that was migrated was created from https://github.com/gardener/gardener/blob/master/example/provider-local/shoot.yaml. 

When I tried to run the test, I've observed multiple different failures that I tried to fixed locally, so maybe I have misinterpreted the graceful failure message and tried to fixed it.

Reply:
I think that the shoot test flavours we use for the `testmachinery` tests have the nginx ingress addon enabled by default. 
Perhaps, enabling it explicitly was done because of trying out to run the test locally during its development? Sadly, I cannot remember the exact reason.

Reply:
Yes, I was attempting to run the test locally (see above for more details).